### DISCUSSION TITLE
Add Redis support for logout purpose

### DISCUSSION THREAD

**Discussion by DanielChuDC** [fastapi-users/fastapi-users#724] [Ideas]
Hi there, thank you for this awesome library. I have read the documentation from v7.0.0 and I notice there has no logout mechanism build in yet.

May I know if this library plan for support logout function? If so, how about add a Redis support for JWT logout?

**Comment by frankie567** [comment]
Currently, there is no JWT invalidation mechanism; it expires by itself following its own lifetime.

I'm well aware that it's not ideal. I'm thinking about implementing a more traditional session management class which would save the token in database.

**Reply by DanielChuDC** [reply]
Thank you for replying! @frankie567 . Understand that session management class may not good when deal with scaling. May I know what kind of database you are considering?

**Reply by frankie567** [reply]
Probably the same set as the ones we support right now: SQL and MongoDB.

That said, if you really need a custom authentication backend with a Redis for session invalidation, it should be pretty straightfoward to implement it in your own project: all you need is to [sub-class `BaseAuthentication`](https://github.com/fastapi-users/fastapi-users/blob/master/fastapi_users/authentication/base.py) and implement each methods. You can very well add some Redis operations inside those methods.

**Reply by DanielChuDC** [reply]
Sound good. Thanks @frankie567 !
